### PR TITLE
v1.9.0-beta.1

### PR DESCRIPTION
**Pull request by ryanhamley**
## Launch Checklist

 - [ ] briefly describe the changes in this PR
 - [ ] include before/after visuals or gifs if this PR includes visual changes
 - [ ] write tests for all new functionality
 - [ ] document any changes to public APIs
 - [ ] post benchmark scores
 - [ ] manually test the debug page
 - [ ] tagged `@mapbox/studio` and/or `@mapbox/map-design-team` if this PR includes style spec or visual changes
 - [ ] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [ ] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [ ] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog></changelog>`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,29 @@
...
+
+## Features and improvements
+* Map API functions such as `easeTo` and `flyTo` now support `padding: PaddingOptions` which lets developers shift the center of perspective for a map when building floating sidebars ([#8638](https://github.com/mapbox/mapbox-gl-js/pull/8638))
+* Introduce `within` expression ([#9352](https://github.com/mapbox/mapbox-gl-js/pull/9352))
+
+### 🍏 Improvements
+* Add changelog policy to contribution guidelines ([#9378](https://github.com/mapbox/mapbox-gl-js/pull/9378))
+* Reduce bundle size and improve visual quality of `showTileBoundaries` debug text ([#9267](https://github.com/mapbox/mapbox-gl-js/pull/9267))
+* Clarify differences in `symbol-sort-key` with and without overlap ([#9379](https://github.com/mapbox/mapbox-gl-js/pull/9379))
+* Add browser testing with Selenium/WebDriver ([#9245](https://github.com/mapbox/mapbox-gl-js/pull/9245))
```
Comment (arindam1993) [inline CHANGELOG.md]: Does this need changelong? I was thinking we can skip this

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+* Slightly improve GPU memory footprint ([#9377](https://github.com/mapbox/mapbox-gl-js/pull/9377))
+* Results from `queryRenderedFeatures` now have evaluated property values rather than raw expressions ([#9198](https://github.com/mapbox/mapbox-gl-js/pull/9198))
+* `LngLatBounds.extend` now accepts `LngLatLike` and `LngLatBoundsLike` as well. ([#9293](https://github.com/mapbox/mapbox-gl-js/pull/9293))
+* Clarify GeolocateControl interaction states ([#9324](https://github.com/mapbox/mapbox-gl-js/pull/9324))
```
Comment (arindam1993) [inline CHANGELOG.md]: This looks like a documentation change, do we want to state it in the changelog?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,29 @@
...
+
+### 🐞 Bug fixes
+* Allow needle argument to `in` expression to be false ([#9295](https://github.com/mapbox/mapbox-gl-js/pull/9295))
+* Clear source tiles when removing a layer ([#9305](https://github.com/mapbox/mapbox-gl-js/pull/9305))
```
Comment (arindam1993) [inline CHANGELOG.md]: Looks like this isn't exactly describing the change.
```suggestion
* Fix exception thrown when trying to set feature-state for a layer that has been removed, fixes [#8634](https://github.com/mapbox/mapbox-gl-js/issues/8634). ([#9305](https://github.com/mapbox/mapbox-gl-js/pull/9305))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,29 @@
+## 1.9.0
+
+## Features and improvements
+* Map API functions such as `easeTo` and `flyTo` now support `padding: PaddingOptions` which lets developers shift the center of perspective for a map when building floating sidebars ([#8638](https://github.com/mapbox/mapbox-gl-js/pull/8638))
+* Introduce `within` expression ([#9352](https://github.com/mapbox/mapbox-gl-js/pull/9352))
+
+### 🍏 Improvements
+* Add changelog policy to contribution guidelines ([#9378](https://github.com/mapbox/mapbox-gl-js/pull/9378))
+* Reduce bundle size and improve visual quality of `showTileBoundaries` debug text ([#9267](https://github.com/mapbox/mapbox-gl-js/pull/9267))
+* Clarify differences in `symbol-sort-key` with and without overlap ([#9379](https://github.com/mapbox/mapbox-gl-js/pull/9379))
```
Comment (arindam1993) [inline CHANGELOG.md]: Also a docs change

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,29 @@
+## 1.9.0
+
+## Features and improvements
+* Map API functions such as `easeTo` and `flyTo` now support `padding: PaddingOptions` which lets developers shift the center of perspective for a map when building floating sidebars ([#8638](https://github.com/mapbox/mapbox-gl-js/pull/8638))
+* Introduce `within` expression ([#9352](https://github.com/mapbox/mapbox-gl-js/pull/9352))
+
+### 🍏 Improvements
+* Add changelog policy to contribution guidelines ([#9378](https://github.com/mapbox/mapbox-gl-js/pull/9378))
```
Comment (arindam1993) [inline CHANGELOG.md]: also a docs change

File: CHANGELOG.md (hunk 1 above)
Comment (karimnaaji) [inline CHANGELOG.md]: nit: Same for this one?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,29 @@
...
+* Reduce bundle size and improve visual quality of `showTileBoundaries` debug text ([#9267](https://github.com/mapbox/mapbox-gl-js/pull/9267))
```
Comment (chloekraw) [inline CHANGELOG.md]: since this is a debug mode improvement, I think it can be moved lower

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,29 @@
...
+* Clarify differences in `symbol-sort-key` with and without overlap ([#9379](https://github.com/mapbox/mapbox-gl-js/pull/9379))
+* Add browser testing with Selenium/WebDriver ([#9245](https://github.com/mapbox/mapbox-gl-js/pull/9245))
+* Slightly improve GPU memory footprint ([#9377](https://github.com/mapbox/mapbox-gl-js/pull/9377))
+* Results from `queryRenderedFeatures` now have evaluated property values rather than raw expressions ([#9198](https://github.com/mapbox/mapbox-gl-js/pull/9198))
```
Comment (asheemmamoowala) [inline CHANGELOG.md]: This is a significant improvement for `queryRenderedFeatures`, what do you think about moving it to the `Features and Improvements` section?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,29 @@
...
+* `LngLatBounds.extend` now accepts `LngLatLike` and `LngLatBoundsLike` as well. ([#9293](https://github.com/mapbox/mapbox-gl-js/pull/9293))
+* Clarify GeolocateControl interaction states ([#9324](https://github.com/mapbox/mapbox-gl-js/pull/9324))
+
+### 🐞 Bug fixes
+* Allow needle argument to `in` expression to be false ([#9295](https://github.com/mapbox/mapbox-gl-js/pull/9295))
+* Clear source tiles when removing a layer ([#9305](https://github.com/mapbox/mapbox-gl-js/pull/9305))
+* Fix a bug where maps were not displaying inside elements with `dir=rtl`. ([#9332](https://github.com/mapbox/mapbox-gl-js/pull/9332))
+* Fix a rendering error for very old versions of Chrome (ca. 2016) where text would appear much bigger than intended ([#9349](https://github.com/mapbox/mapbox-gl-js/pull/9349))
+* Fix a bug where `icon-image` expression that evaluates to `''` produced a warning. ([#9380](https://github.com/mapbox/mapbox-gl-js/pull/9380))
```
Comment (asheemmamoowala) [inline CHANGELOG.md]:
```suggestion
* Fix a bug where `icon-image` expression that evaluates to an empty string (`''`) produced a warning. ([#9380](https://github.com/mapbox/mapbox-gl-js/pull/9380))
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,29 @@
...
+* Prevent exception resulting from `line-dash-array` of empty length ([#9385](https://github.com/mapbox/mapbox-gl-js/pull/9385))
+* Improve scaling of patterns used in `line-pattern` on all device resolutions and pixel ratios ([#9266](https://github.com/mapbox/mapbox-gl-js/pull/9266))
```
Comment (asheemmamoowala) [inline CHANGELOG.md]: Recommend moving this to the top section as well.

[the remark this answers is not in the document]
Comment (chloekraw) [reply CHANGELOG.md]: @karimnaaji: let's call it a visual change?

[the remark this answers is not in the document]
Comment (chloekraw) [reply CHANGELOG.md]: Also, I suggest linking to the ticket with a description under the release, like we did for 1.5.1: https://github.com/mapbox/mapbox-gl-js/releases/tag/v1.5.1. Suggested wording:

> This release fixes several longstanding bugs and unintentional rendering behavior with `line-pattern`. The fixes come with a visual change to how patterns added with `line-pattern` scale. Previously, patterns that became larger than the line would be clipped, sometimes distorting the pattern, particularly on mobile and retina devices. Now, the pattern will be scaled to fit under all circumstances. #9266 showcases an example of the visual differences. For more information and to provide feedback on this change, see #9394.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,30 @@
...
+
+This release also fixes several longstanding bugs and unintentional rendering behavior with line-pattern. The fixes come with a visual change to how patterns added with line-pattern scale. Previously, patterns that became larger than the line would be clipped, sometimes distorting the pattern, particularly on mobile and retina devices. Now, the pattern will be scaled to fit under all circumstances. [#9266](https://github.com/mapbox/mapbox-gl-js/pull/9266) showcases examples of the visual differences. For more information and to provide feedback on this change, see [#9394](https://github.com/mapbox/mapbox-gl-js/pull/9394).
+
+### ✨ Features
+* Add `within` expression for testing whether an evaluated feature lies within a given GeoJSON object ([#9352](https://github.com/mapbox/mapbox-gl-js/pull/9352)). For example:<br>
+`"icon-opacity": ["case", ["==", ["within", "some-polygon"], true], 1,
+["==", ["within", "some-polygon"], false], 0]`
+* Map API functions such as `easeTo` and `flyTo` now support `padding: PaddingOptions` which lets developers shift the center of perspective for a map when building floating sidebars ([#8638](https://github.com/mapbox/mapbox-gl-js/pull/8638))
```
Comment (chloekraw) [inline CHANGELOG.md]: nit: which lets developers shift a map's center of perspective when building floating sidebars

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,30 @@
...
+
+### 🍏 Improvements
+* Results from `queryRenderedFeatures` now have evaluated property values rather than raw expressions ([#9198](https://github.com/mapbox/mapbox-gl-js/pull/9198))
+* Improve scaling of patterns used in `line-pattern` on all device resolutions and pixel ratios. ([#9266](https://github.com/mapbox/mapbox-gl-js/pull/9266))
```
Comment (chloekraw) [inline CHANGELOG.md]: nit: remove period

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,30 @@
...
+* Slightly improve GPU memory footprint ([#9377](https://github.com/mapbox/mapbox-gl-js/pull/9377))
+* `LngLatBounds.extend` is more flexible because it now accepts objects with `lat` and `lon` properties as well as arrays of coordinates ([#9293](https://github.com/mapbox/mapbox-gl-js/pull/9293))
+* Reduce bundle size and improve visual quality of `showTileBoundaries` debug text ([#9267](https://github.com/mapbox/mapbox-gl-js/pull/9267))
+
+### 🐞 Bug fixes
+* Correctly adjust patterns added with `addImage(id, image, pixelRatio)` by the asset pixel ratio, not the device pixel ratio ([#9372](https://github.com/mapbox/mapbox-gl-js/pull/9372))
+* Allow needle argument to `in` expression to be false ([#9295](https://github.com/mapbox/mapbox-gl-js/pull/9295))
+* Fix exception thrown when trying to set `feature-state` for a layer that has been removed, fixes [#8634](https://github.com/mapbox/mapbox-gl-js/issues/8634) ([#9305](https://github.com/mapbox/mapbox-gl-js/pull/9305))
+* Fix a bug where maps were not displaying inside elements with `dir=rtl`. ([#9332](https://github.com/mapbox/mapbox-gl-js/pull/9332))
```
Comment (chloekraw) [inline CHANGELOG.md]: nit: remove period

**Pull request merged**